### PR TITLE
Add some examples of widgets that support visual density

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This adds a list of widgets that respond to visual density changes to the API docs for `VisualDensity`.

## Tests

- Doc only change.

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -2095,6 +2095,21 @@ class _FifoCache<K, V> {
 /// the list. For chips, it only affects the vertical size, not the horizontal
 /// size.
 ///
+/// Here are some examples of widgets that respond to density changes:
+///  - [Checkbox]
```
Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart:2099]: you'll need a blank line here I believe or it'll just become a big paragraph

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -2095,6 +2095,21 @@ class _FifoCache<K, V> {
...
+///  - [MaterialButton]
+///  - [OutlineButton]
+///  - [OutlinedButton]
+///  - [Radio]
+///  - [RawMaterialButton]
+///  - [TextButton]
+///
 /// See also:
 ///
 ///  * [ThemeData.visualDensity], where this property is used to specify the base
```
Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart:2115]: looks like we use `*` here for bullets, maybe we should use `*` above too?

File: packages/flutter/lib/src/material/theme_data.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/theme_data.dart:2099]: <*shakes fist*> darn you markdown!

**Comment by Hixie** [comment]
test-exempt: only affects comments

That said, I wonder if we need a markdown linter or something. :-)

File: packages/flutter/lib/src/material/theme_data.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/theme_data.dart:2116]: Done.

**Comment by gspencergoog** [comment]
> That said, I wonder if we need a markdown linter or something. :-)

Yeah, it would be nice if `dartdoc` provided that service for us, since it's really the one doing the processing there.

**Review by Piinks — APPROVED** [review]
LGTM

**Pull request merged**